be/src/runtime/thread_context.h

```diff
@@ -460,17 +478,22 @@ class ScopeSkipMemoryCheck {
 // Mem Hook to consume thread mem tracker
 #define CONSUME_THREAD_MEM_TRACKER_BY_HOOK(size)           \
     do {                                                   \
+        allocator_detect(size);                            \
         if (doris::use_mem_hook) {                         \
             doris::thread_context()->consume_memory(size); \
         }                                                  \
     } while (0)
+
 #define RELEASE_THREAD_MEM_TRACKER_BY_HOOK(size) CONSUME_THREAD_MEM_TRACKER_BY_HOOK(-size)
+
 #define CONSUME_THREAD_MEM_TRACKER_BY_HOOK_WITH_FN(size_fn, ...)           \
     do {                                                                   \
+        allocator_detect(size_fn(__VA_ARGS__));                            \
```

Comment:
```
#ifndef NDEBUG
allocator_detect(size_fn(__VA_ARGS__));
#endif
```
avoid unnecessary function calls `size_fn(__VA_ARGS__)` and `allocator_detect`

Reply:
function all will be erased with -O3